omod/src/test/java/org/openmrs/module/owa/web/controller/AddAppControllerTest.java

```diff
@@ -46,11 +52,34 @@ public void restoreOriginalMessageSourceService() {
 	 * Test of upload method, of class AddAppController.
 	 */
 	@Test
-	public void testUpload() throws Exception {
-		System.out.println("upload");
+	public void testUploadofNotaZipArchieve() throws Exception {
+		HttpServletRequest request = new MockHttpServletRequest(new MockServletContext(), "POST", "/module/owa/addApp.htm");
+		URL url = OpenmrsClassLoader.getInstance().getResource("testing");
+		MockMultipartFile multifile = new MockMultipartFile("testFile", "testing", null, new FileInputStream(url.getFile()));
+		AddAppController controller = (AddAppController) applicationContext.getBean("addAppController");
+		controller.upload(multifile, request);
+		Assert.assertEquals(request.getSession().getAttribute("errormessage"), "owa.not_a_zip");
+	}
+	
+	@Test
+	public void testUploadofEmptyFile() throws Exception {
+		HttpServletRequest request = new MockHttpServletRequest(new MockServletContext(), "POST", "/module/owa/addApp.htm");
+		URL url = OpenmrsClassLoader.getInstance().getResource("file");
+		MockMultipartFile multifile = new MockMultipartFile("testFile", "file", null, new FileInputStream(url.getFile()));
+		AddAppController controller = (AddAppController) applicationContext.getBean("addAppController");
+		controller.upload(multifile, request);
+		Assert.assertEquals(request.getSession().getAttribute("errormessage"), "owa.blank_zip");
+	}
+	
+	@Test
+	public void testUploadofZipFilewithoutManifest() throws Exception {
 		HttpServletRequest request = new MockHttpServletRequest(new MockServletContext(), "POST", "/module/owa/addApp.htm");
-		AddAppController instance = new AddAppController();
-		String expResult = "";
+		URL url = OpenmrsClassLoader.getInstance().getResource("ZipFilewithoutmanifest.zip");
+		MockMultipartFile multifile = new MockMultipartFile("testFile", "ZipFilewithoutmanifest.zip", null,
+		        new FileInputStream(url.getFile()));
+		AddAppController controller = (AddAppController) applicationContext.getBean("addAppController");
+		controller.upload(multifile, request);
```

Comment:
This line throws Zipexception (error in opening Zip File) while it works fine when tested manually So i am not sure where is the error coming from Can i get some help on this :)